client/client.go

```diff
@@ -207,6 +218,23 @@ func (c *Client) processSupervisorRequests() {
 	wg.Wait()
 }
 
+func (c *Client) sendActionResults() {
+	for ar := range c.runner.ActionReady() {
+		var errMessage string
+		if ar.Error != nil {
+			errMessage = ar.Error.Error()
+		}
+
+		c.channel.SendRequest(&agentpb.ActionResultRequest{
+			ActionId: ar.ID,
+			Done:     ar.Error == nil,
```

Comment:
Action is done when it is done, not when it is done without error (or with error as in the previous version of this line).